### PR TITLE
Db index refactoring

### PR DESCRIPTION
**Pull request by abolinbsu**

### CODE REVIEW

File: libraries/chainbase/CMakeLists.txt [hunk 1]
```diff
@@ -62,7 +62,8 @@ endif()
 
 
 file(GLOB HEADERS "include/*.hpp")
-add_library( chainbase src/chainbase.cpp ${HEADERS} )
+file(GLOB SRC_FILES src/*.cpp)
```
Comment (Andrei-Masilevich) [inline libraries/chainbase/CMakeLists.txt]: I would be better to delete src folder:
git mv src/file.cpp ./

File: libraries/chainbase/include/chainbase/abstract_interfaces.hpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    virtual ~abstract_undo_session(){};
+
+    virtual void push() = 0;
+};
+
+using abstract_undo_session_ptr = std::unique_ptr<abstract_undo_session>;
+using abstract_undo_session_list = std::vector<abstract_undo_session_ptr>;
+
+//------------------------------------------------------------------------------------------------------//
+struct abstract_generic_index
```
Comment (Andrei-Masilevich) [inline libraries/chainbase/include/chainbase/abstract_interfaces.hpp]: rename to abstract_generic_index_i

File: libraries/chainbase/include/chainbase/abstract_interfaces.hpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+{
+    virtual ~abstract_undo_session(){};
+
+    virtual void push() = 0;
+};
+
+using abstract_undo_session_ptr = std::unique_ptr<abstract_undo_session>;
+using abstract_undo_session_list = std::vector<abstract_undo_session_ptr>;
+
+//------------------------------------------------------------------------------------------------------//
```
Comment (alexplevako) [inline libraries/chainbase/include/chainbase/abstract_interfaces.hpp]: remove this comment

File: libraries/chainbase/include/chainbase/chainbase.hpp [hunk 4]
```diff
@@ -1,771 +1,25 @@
...
-    }
+    bfs::path _data_dir;
 
-    uint32_t current_lock_num()
-    {
-        return _current_lock;
-    }
+    std::unique_ptr<bip::managed_mapped_file> _meta;
 
 private:
```
Comment (alexplevako) [inline libraries/chainbase/include/chainbase/chainbase.hpp]: remove ```private:```

**Comment by alexplevako** [comment]
move chainbase test to the chain_test project.

**Pull request merged**